### PR TITLE
Change to deploy hosting to Firebase

### PR DESCRIPTION
**Pull request by joeySchentrup**
* DO has a cert that expired because it was not renewed. We don't want to have to manage a VM or certs so let Google do that for us
* Allows us to have a testing site and deploy to production only on release tag
* Allows us to manage functions and DB rules directly in the repo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +35,10 @@
         "start": "react-scripts start",
         "build": "react-scripts build",
         "test": "react-scripts test",
-        "serve": "serve -s build -l 3000"
+        "build:functions": "yarn global add tslint@5.12.0 && yarn global add typescript@3.4 && cd functions && npm install",
```
Comment (RyanClementsHax) [inline package.json]: why are we installing global dependencies? cant we just include them as part of the project's dependencies?

File: storage.rules [hunk 2]
```diff
@@ -0,0 +1,7 @@
+service firebase.storage {
```
Comment (RyanClementsHax) [inline storage.rules]: do we need this?

File: package.json (hunk 1 above)
Comment (joeySchentrup) [reply package.json]: I tried very very hard to not need this but for some reason I could not get it to work. The dependencies would be installed locally but the commands could not be found on the travis vm when it ran.

File: storage.rules (hunk 2 above)
Comment (joeySchentrup) [reply storage.rules]: We currently do not need it. It does however let us configure it easily if we ever do need it in the future plus keeps the configuration in the repo so it cannot be changed without us knowing it.

**Pull request merged**